commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/http5/Http5FileObject.java

```diff
@@ -94,7 +93,7 @@ protected Http5FileObject(final AbstractFileName name, final FS fileSystem,
         final FileSystemOptions fileSystemOptions = fileSystem.getFileSystemOptions();
         urlCharset = builder.getUrlCharset(fileSystemOptions);
         final String pathEncoded = ((GenericURLFileName) name).getPathQueryEncoded(getUrlCharset());
-        internalURI = URIUtils.resolve(fileSystem.getInternalBaseURI(), pathEncoded);
+        internalURI = fileSystem.getInternalBaseURI().resolve(pathEncoded);
```

Comment:
`URIUtils.resolve()` from HttpClient5.1  has an issue of escaping host symbols that is not working properly with IPv6 hosts. HttpClient4 and native `Java URI.resolve()` don't have this problem.